google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/facet/standard/AppEngineStandardWebIntegration.java

```diff
@@ -111,34 +111,51 @@ public void registerFrameworkInModel(FrameworkSupportModel model,
   }
 
   private void setupDeployRunConfiguration(@NotNull Module module) {
-    AppEngineCloudType serverType =
-        ServerType.EP_NAME.findExtension(AppEngineCloudType.class);
-    RemoteServer<AppEngineServerConfiguration> server
-        = ContainerUtil.getFirstItem(RemoteServersManager.getInstance().getServers(serverType));
-
-    DeployToServerConfigurationType configurationType
-        = DeployToServerConfigurationTypesRegistrar.getDeployConfigurationType(serverType);
     RunManager runManager = RunManager.getInstance(module.getProject());
-    ConfigurationFactoryEx factory = configurationType.getFactory();
-    RunnerAndConfigurationSettings settings = runManager.createRunConfiguration(
-        configurationType.getDisplayName(), factory);
-    DeployToServerRunConfiguration<?, ?> runConfiguration
-        = (DeployToServerRunConfiguration<?, ?>)settings.getConfiguration();
-
-    if (server != null) {
-      runConfiguration.setServerName(server.getName());
-    }
 
-    runManager.addConfiguration(settings, false /*isShared*/);
+    boolean hasExistingDeployConfiguration = runManager.getAllConfigurationsList()
```

Comment:
I think its cleaner to leave it as is. I _could_ do something like this:

```
    RunManagerImpl runManager = RunManagerImpl.getInstanceImpl(module.getProject());
    ConfigurationType configType = runManager.getConfigurationType(AppEngineCloudType.ID);

    boolean hasExistingDeployConfiguration = configType == null ||
        runManager.getConfigurationsList(configType)
            .stream()
            .map(config -> ((DeployToServerRunConfiguration) config).getServerType())
            .anyMatch(serverType -> serverType instanceof AppEngineCloudType);
```
But since configType is nullable it makes the code a little uglier.